master_server.py

```diff
@@ -45,12 +47,16 @@ def create_servers(silent=False, debug_mode=False):
 
 def main(args):
     """Master server main function."""
-    servers, has_ui = create_servers(silent=args.silent,
+    servers, has_ui = create_servers(server_id=args.server_id,
+                                     silent=args.silent,
                                      debug_mode=args.debug_mode)
     threads = [
         threading.Thread(target=server.serve_forever)
         for server in servers
     ]
+    cache = Cache(server_id=args.server_id, debug_mode=args.debug_mode, log_to_file=True,
+                log_to_console=not args.silent, log_to_window=False)
+    threads.append(threading.Thread(target=cache.maintain_connection))
```

Comment:
If the only way to start the "cache" server is by running the `master_server.py` that seems wrong by design.